### ISSUE TITLE
upgrade sratoolkit fails

### ISSUE THREAD

**Issue by alexstaj** [Linuxbrew/homebrew-core#5323]
```
tha@bioinfo:~$ brew upgrade
==> Upgrading 1 outdated package, with result:
sratoolkit 2.8.2-5
==> Upgrading sratoolkit 
==> Using Homebrew-provided fortran compiler.
This may be changed by setting the FC environment variable.
==> Downloading https://github.com/ncbi/sra-tools/archive/2.8.2-5.tar.gz
Already downloaded: /home/tha/.cache/Homebrew/sratoolkit-2.8.2-5.tar.gz
==> Downloading https://github.com/ncbi/ngs/archive/1.3.0.tar.gz
Already downloaded: /home/tha/.cache/Homebrew/sratoolkit--ngs-sdk-1.3.0.tar.gz
==> ./configure --prefix=/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ngs-sdk-prefix --build=/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ng
==> make
==> make install
==> Downloading https://github.com/ncbi/ncbi-vdb/archive/2.8.2-2.tar.gz
Already downloaded: /home/tha/.cache/Homebrew/sratoolkit--ncbi-vdb-2.8.2-2.tar.gz
==> ./configure --prefix=/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-prefix --with-ngs-sdk-prefix=/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-t
==> make
==> ./configure --prefix=/home/tha/.linuxbrew/Cellar/sratoolkit/2.8.2-5 --with-ngs-sdk-prefix=/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ngs-sdk-pref
==> make install
Last 15 lines from /home/tha/.cache/Homebrew/Logs/sratoolkit/07.make:
gcc -static-libgcc -o /tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/bin/helicos-load.2.8.2 -DNDEBUG -m64 pstring.o common-xml.o experiment-xml.o run-xml.o loader-file.o loader-fmt.o sra-writer.o loader.o writer-helicos.o helicos-fmt.o -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/lib -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/ilib -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ngs-sdk-prefix/lib64 -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-build/ncbi-vdb/linux/gcc/x86_64/rel/lib -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-build/ncbi-vdb/linux/gcc/x86_64/rel/ilib -Wl,-Bstatic -lkapp -ltk-version -lload -lncbi-wvdb -lkxfs -lkxml -Wl,-Bdynamic -ldl -lpthread -lxml2 -Wl,-Bdynamic -lm
gcc -c   -DNDEBUG -m64  -std=gnu99 -fgnu89-inline -DLINUX -DUNIX -D_GNU_SOURCE -D_REENTRANT -D_FILE_OFFSET_BITS=64 -DPKGNAME=linux64 -D_ARCH_BITS=__SIZEOF_POINTER__*__CHAR_BIT__ -DLIBPREFIX=lib -DSHLIBEXT=so  -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/interfaces/override -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/tools/sra-sort/linux -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/tools/sra-sort/unix -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/tools/sra-sort -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/ -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/cc/gcc/x86_64 -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/cc/gcc -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/os/linux -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/os/unix -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/ext -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ngs-sdk-prefix/include -I. -MD -o xcheck-ref-align.o -O3 -Wno-variadic-macros -fno-strict-aliasing -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -Wall /tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/tools/sra-sort/xcheck-ref-align.c -D__mod_name__=tools/sra-sort -D__file_name__=xcheck-ref-align -D__file_ext__=c
ln -s helicos-load.2.8.2 helicos-load.2
ln -s helicos-load.2 helicos-load
gcc -c   -DNDEBUG -m64   -DLINUX -DUNIX -D_GNU_SOURCE -D_REENTRANT -D_FILE_OFFSET_BITS=64 -DPKGNAME=linux64 -D_ARCH_BITS=__SIZEOF_POINTER__*__CHAR_BIT__ -DLIBPREFIX=lib -DSHLIBEXT=so  -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/interfaces/override -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/tools/sra-load/linux -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/tools/sra-load/unix -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/tools/sra-load -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/ -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/cc/gcc/x86_64 -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/cc/gcc -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/os/linux -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/os/unix -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/ext -I/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ngs-sdk-prefix/include -I. -MD -o pacbio-fmt.o -O3 -Wno-variadic-macros -fno-strict-aliasing -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -Wall /tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/tools/sra-load/pacbio-fmt.c -D__mod_name__=tools/sra-load -D__file_name__=pacbio-fmt -D__file_ext__=c
gcc -static-libgcc -o /tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/bin/pacbio-loadxml.2.8.2 -DNDEBUG -m64 pstring.o common-xml.o experiment-xml.o run-xml.o loader-file.o loader-fmt.o sra-writer.o loader.o pacbio-fmt.o -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/lib -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/ilib -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ngs-sdk-prefix/lib64 -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-build/ncbi-vdb/linux/gcc/x86_64/rel/lib -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-build/ncbi-vdb/linux/gcc/x86_64/rel/ilib -Wl,-Bstatic -lkapp -ltk-version -lload -lncbi-wvdb -lkxfs -lkxml -Wl,-Bdynamic -ldl -lpthread -lxml2 -Wl,-Bdynamic -lm
gcc -static-libgcc -o /tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/bin/sra-sort.2.8.2 -DNDEBUG -m64 caps.o mem.o membank.o paged-membank.o paged-mmapbank.o except.o idx-mapping.o map-file.o col-pair.o row-set.o simple-row-set.o mapping-row-set.o sorting-row-set.o meta-pair.o dir-pair.o tbl-pair.o db-pair.o glob-poslen.o poslen-col-pair.o ref-alignid-col.o buff-writer.o id-mapper-col.o capture-first-half-aligned.o csra-tbl.o csra-pair.o run.o sra-sort.o xcheck-ref-align.o -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/lib -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/ilib -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ngs-sdk-prefix/lib64 -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-build/ncbi-vdb/linux/gcc/x86_64/rel/lib -L/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/ncbi-vdb-build/ncbi-vdb/linux/gcc/x86_64/rel/ilib -Wl,-Bstatic -lkapp -ltk-version -lncbi-wvdb -Wl,-Bdynamic -ldl -lpthread -lxml2 -Wl,-Bdynamic -lm
ln -s pacbio-loadxml.2.8.2 pacbio-loadxml.2
ln -s pacbio-loadxml.2 pacbio-loadxml
ln -s sra-sort.2.8.2 sra-sort.2
ln -s sra-sort.2 sra-sort
/tmp/sratoolkit-20171227-1539-1c9r6sb/sra-tools-2.8.2-5/build/Makefile.targets:66: recipe for target 'tools_std' failed
make[1]: *** [tools_std] Error 2
Makefile:66: recipe for target 'install' failed
make: *** [install] Error 2

READ THIS: https://github.com/Linuxbrew/brew/blob/master/docs/Troubleshooting.md#troubleshooting
Please do not report this issue to Homebrew/brew or Homebrew/core, which support macOS only.



```

**Comment by iMichka** [comment]
Hi. Thanks for reporting this.

The actual build error is (test in a local docker container):
```
gcc -c   -DNDEBUG -m64  -std=gnu99 -fgnu89-inline -DLINUX -DUNIX -D_GNU_SOURCE -D_REENTRANT -D_FILE_OFFSET_BITS=64 -DPKGNAME=linux64 -D_ARCH_BITS=__SIZEOF_POINTER__*__CHAR_BIT__ -DLIBPREFIX=lib -DSHLIBEXT=so  -I/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/interfaces/override -I/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/tools/sra-sort/linux -I/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/tools/sra-sort/unix -I/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/tools/sra-sort -I/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/ -I/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/cc/gcc/x86_64 -I/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/cc/gcc -I/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/os/linux -I/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/os/unix -I/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/ncbi-vdb-source/interfaces/ext -I/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/ngs-sdk-prefix/include -I. -MD -o id-mapper-col.o -O3 -Wno-variadic-macros -fno-strict-aliasing -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -Wall /tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/tools/sra-sort/id-mapper-col.c -D__mod_name__=tools/sra-sort -D__file_name__=id-mapper-col -D__file_ext__=c
/home/linuxbrew/.linuxbrew/lib/libhdf5.a(H5Z.o): In function `H5Z__init_package':
H5Z.c:(.text+0x6a1): undefined reference to `SZ_encoder_enabled'
/home/linuxbrew/.linuxbrew/lib/libhdf5.a(H5Zszip.o): In function `H5Z_filter_szip':
H5Zszip.c:(.text+0x153): undefined reference to `SZ_BufftoBuffCompress'
H5Zszip.c:(.text+0x1e1): undefined reference to `SZ_BufftoBuffDecompress'
collect2: error: ld returned 1 exit status
/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/tools/pacbio-load/Makefile:117: recipe for target '/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/bin/pacbio-load' failed
make[4]: *** [/tmp/sratoolkit-20171228-32191-1aeja5d/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/bin/pacbio-load] Error 1
Makefile:59: recipe for target 'std' failed
```

**Comment by iMichka** [comment]
Similar situation here: https://github.com/PacificBiosciences/blasr/issues/4#issuecomment-45689646
This may help fixing this.

**Comment by alexstaj** [comment]
hi thanks for the response but how do I implement the make static argument with regards to brew upgrade sratoolkit?

```
When compiling statically -static, you must link szip -lsz. To work around, run make STATIC=.
```

**Comment by andrewyatz** [comment]
I've seen a similar issue with sratoolkit just now

```
gcc -static-libstdc++ -static-libgcc -o /tmp/sratoolkit-20180105-50520-o4uj0v/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/bin/sra-sort.2.8.2 -DNDEBUG -m64 caps.o mem.o membank.o paged-membank.o paged-mmapbank.o except.o idx-mapping.o map-file.o col-pair.o row-set.o simple-row-set.o mapping-row-set.o sorting-row-set.o meta-pair.o dir-pair.o tbl-pair.o db-pair.o glob-poslen.o poslen-col-pair.o ref-alignid-col.o buff-writer.o id-mapper-col.o capture-first-half-aligned.o csra-tbl.o csra-pair.o run.o sra-sort.o xcheck-ref-align.o -L/tmp/sratoolkit-20180105-50520-o4uj0v/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/lib -L/tmp/sratoolkit-20180105-50520-o4uj0v/sra-tools-2.8.2-5/sra-tools-build/sra-tools/linux/gcc/x86_64/rel/ilib -L/tmp/sratoolkit-20180105-50520-o4uj0v/sra-tools-2.8.2-5/ngs-sdk-prefix/lib64 -L/tmp/sratoolkit-20180105-50520-o4uj0v/sra-tools-2.8.2-5/ncbi-vdb-build/ncbi-vdb/linux/gcc/x86_64/rel/lib -L/tmp/sratoolkit-20180105-50520-o4uj0v/sra-tools-2.8.2-5/ncbi-vdb-build/ncbi-vdb/linux/gcc/x86_64/rel/ilib -Wl,-Bstatic -lkapp -ltk-version -lncbi-wvdb -Wl,-Bdynamic -ldl -lpthread -lxml2 -Wl,-Bdynamic -lm
ln -s sra-sort.2.8.2 sra-sort.2
ln -s sra-sort.2 sra-sort
make[1]: *** [tools_std] Error 2
make: *** [install] Error 2
```

Same sort of area but little more information than what you've seen here. The `.cc` log files bring little extra information

**Comment by alexstaj** [comment]
Thanks for the update Andrew.  Any idea what arguments I need to add to
brew install sratoolkit to get it to install properly?

**Comment by andrewyatz** [comment]
Not at all sorry. I'm totally lost as to why it's happened as well as the last version of the sratoolkit (2.8.2-1_1 for me) was fine.

**Comment by iMichka** [comment]
Actually it is weird that it tries to make a static linkage here.

Can somebody check if the szip and the hdf5 installations have .so files (and not only .a files)?

**Comment by andrewyatz** [comment]
I can check on mine. Lots of .so files

```
$ ls $(brew --prefix szip)/lib
libsz.a  libsz.so  libsz.so.2  libsz.so.2.0.0
$ ls $(brew --prefix hdf5)/lib
libdynlib1.so	 libdynlibdiff.so  libhdf5_cpp.a	   libhdf5_fortran.so	       libhdf5_hl_cpp.so	  libhdf5hl_fortran.so.100	libhdf5.so
libdynlib2.so	 libdynlibdump.so  libhdf5_cpp.so	   libhdf5_fortran.so.100      libhdf5_hl_cpp.so.100	  libhdf5hl_fortran.so.100.0.1	libhdf5.so.101
libdynlib3.so	 libdynlibls.so    libhdf5_cpp.so.101	   libhdf5_fortran.so.100.1.0  libhdf5_hl_cpp.so.100.1.0  libhdf5_hl.so			libhdf5.so.101.0.0
libdynlib4.so	 libdynlibvers.so  libhdf5_cpp.so.101.0.0  libhdf5_hl.a		       libhdf5hl_fortran.a	  libhdf5_hl.so.100
libdynlibadd.so  libhdf5.a	   libhdf5_fortran.a	   libhdf5_hl_cpp.a	       libhdf5hl_fortran.so	  libhdf5_hl.so.100.0.1
```

**Comment by iMichka** [comment]
I started a build here just to see what happens: #5399

**Comment by andrewyatz** [comment]
I don't know if it's a good thing or not but the Travis build has the same issue. I've emailed the maintainers with links to this issue and to the Travis build and will see if they can suggest a solution.

**Comment by iMichka** [comment]
This was fixed by #5616
